src/nspawn/nspawn.c

```diff
@@ -4667,6 +4674,33 @@ static int run_container(
         /* Normally redundant, but better safe than sorry */
         (void) kill(*pid, SIGKILL);
 
+        if (arg_private_network) {
+                /* Move network interfaces back to the parent network namespace. */
+                _cleanup_close_ int parent_netns_fd = -1;
+
+                r = namespace_open(getpid_cached(), NULL, NULL, &parent_netns_fd, NULL, NULL);
+                if (r < 0)
+                        return log_error_errno(r, "Failed to open parent network namespace: %m");
```

Comment:
why not do this in the child? i mean it doesn't really matter, but i'd keep the fd scope short if it's easy